### ISSUE TITLE
[BUG] Directory creation errors

### ISSUE THREAD

**Issue by lenary** [defunkt/rip#59]
I'm getting the following error when i try to use rip:

```
MAC46SE ~: rip install git://github.com/defunkt/grit.git v1.1.1b
fatal: destination path 'grit-1dce6a545b29fd1845f9dc123ead7fa8' already exists and is not an empty directory.
rip: install failed
-> No such file or directory - /Users/same/.rip/rip-packages/grit-1dce6a545b29fd1845f9dc123ead7fa8
```

I don't know what the problem is. I used the gem install, and `rip check` returns `all systems go`. the directories are chmodded to my user, so i don't know what else the problem could be

**Comment by defunkt** [comment]
What version of Ruby and OS are you using?

**Comment by lenary** [comment]
ruby -v:
`ruby 1.8.6 (2008-08-11 patchlevel 287) [universal-darwin9.0]`

Mac OS X 10.5.7

**Comment by protocarl** [comment]
I'm getting a similar behavior on a similar configuration.  These tests fail on f375cf0 reflecting the problem.

```
$ ruby --version
ruby 1.8.6 (2008-08-11 patchlevel 287) [universal-darwin9.0]

$ rip list
ripenv: ripdev

fakefs (99880ea)
test-spec-mini.rb (2009-07-23)

$ ruby test/git_test.rb
Loaded suite test/git_test
Started
.E.EE
Finished in 0.502716 seconds.

  1) Error:
test_fails_on_an_unknown_version():
Errno::ENOENT: No such file or directory - /Users/carl/Projects/rip/test/ripdir/rip-packages/simple_c-156d7c2b9cfa0bab0ee4a6f118998e17
    /Users/carl/.rip/active/lib/fakefs.rb:338:in `chdir'
    /Users/carl/.rip/active/lib/fakefs.rb:256:in `chdir'
    /Users/carl/Projects/rip/lib/rip/packages/git_package.rb:62:in `remote_exists?'
    /Users/carl/Projects/rip/lib/rip/packages/git_package.rb:26:in `exists?'
    /Users/carl/Projects/rip/lib/rip/installer.rb:17:in `install'
    /Users/carl/Projects/rip/lib/rip/commands/install.rb:34:in `install'
    test/git_test.rb:21:in `test_fails_on_an_unknown_version'

  2) Error:
test_works_with_a_real_version():
Errno::ENOENT: No such file or directory - /Users/carl/Projects/rip/test/ripdir/rip-packages/simple_c-156d7c2b9cfa0bab0ee4a6f118998e17
    /Users/carl/.rip/active/lib/fakefs.rb:338:in `chdir'
    /Users/carl/.rip/active/lib/fakefs.rb:256:in `chdir'
    /Users/carl/Projects/rip/lib/rip/packages/git_package.rb:62:in `remote_exists?'
    /Users/carl/Projects/rip/lib/rip/packages/git_package.rb:26:in `exists?'
    /Users/carl/Projects/rip/lib/rip/installer.rb:17:in `install'
    /Users/carl/Projects/rip/lib/rip/commands/install.rb:34:in `install'
    test/git_test.rb:26:in `test_works_with_a_real_version'

  3) Error:
test_works_with_an_existing_sha():
Errno::ENOENT: No such file or directory - /Users/carl/Projects/rip/test/ripdir/rip-packages/simple_c-156d7c2b9cfa0bab0ee4a6f118998e17
    /Users/carl/.rip/active/lib/fakefs.rb:338:in `chdir'
    /Users/carl/.rip/active/lib/fakefs.rb:256:in `chdir'
    /Users/carl/Projects/rip/lib/rip/packages/git_package.rb:62:in `remote_exists?'
    /Users/carl/Projects/rip/lib/rip/packages/git_package.rb:26:in `exists?'
    /Users/carl/Projects/rip/lib/rip/installer.rb:17:in `install'
    /Users/carl/Projects/rip/lib/rip/commands/install.rb:34:in `install'
    test/git_test.rb:32:in `test_works_with_an_existing_sha'

5 tests, 2 assertions, 0 failures, 3 errors
```

**Comment by lenary** [comment]
check those with defunkt - he mentioned having checked in some failing tests to do with git :/

mind you, that was with edge, not with 0.0.2

**Comment by defunkt** [comment]
What version of rip are you guys running?

```
$ rip install git://github.com/defunkt/grit.git v1.1.1b
Successfully installed diff-lcs (491fbc0)
Successfully installed mime-types (v1.16)
Successfully installed grit (v1.1.1b)
$ rip --version
Rip 0.0.2
```

I'm aware of the broken tests on edge.

**Comment by protocarl** [comment]
Running a clean install of 0.0.2:

```
$ sudo gem install rip
Building native extensions.  This could take a while...
****************************************************
So far so good...

Rip needs certain env variables to run. We've tried
to install them automatically but may have failed.

Run `rip check` to check the status of your
installation.

Get started: `rip -h` or http://hellorip.com/

****************************************************
Successfully installed rip-0.0.2
1 gem installed
$ rip env create grit
ripenv: created grit
$ rip install git://github.com/defunkt/grit.git v1.1.1b
rip: install failed
-> No such file or directory - /Users/carl/.rip/rip-packages/grit-1dce6a545b29fd1845f9dc123ead7fa8
$ rip --version
Rip 0.0.2
```

I also get the same test failures on this version.

**Comment by lenary** [comment]
```
MAC46SE ~: rip --version
Rip 0.0.2
```

**Comment by defunkt** [comment]
Fix the "directory not found" error when installing git packages. Closed by 2cd750fc51df7a8fcbddcf1249eb9bd8e2d02a50
